docs/configuration-reference/components/azure-arc-onboarding.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Azure Arc onboarding configuration reference for Lokomotive
+weight: 10
+---
+
+## Introduction
+
+Azure Arc offers simplified management, faster app development, and consistent Azure services.
+
+With Azure Arc, you can:
+
+- Centrally manage a wide range of resources including
+[Windows](https://azure.microsoft.com/en-in/campaigns/windows-server/) and
+[Linux](https://azure.microsoft.com/en-in/overview/linux-on-azure/) servers, SQL server,
+[Kubernetes](https://azure.microsoft.com/en-in/services/kubernetes-service/) clusters and [Azure
+services](https://azure.microsoft.com/en-in/services/azure-arc/hybrid-data-services/).
+
+- Establish central visibility in the [Azure portal](https://azure.microsoft.com/en-in/features/azure-portal/)
+and enable multi-environment search with Azure Resource Graph.
+
+- Meet [governance](https://azure.microsoft.com/en-in/solutions/governance/) and compliance standards for
+apps, infrastructure and data with [Azure Policy](https://azure.microsoft.com/en-in/services/azure-policy/).
+
+- Delegate access and manage security policies for resources using role-based access control (RBAC) and [Azure
+Lighthouse](https://azure.microsoft.com/en-in/services/azure-lighthouse/).
+
+- Organise and inventory assets through a variety of Azure scopes, such as management groups, subscriptions,
+resource groups and tags.
+
+This component onboards or removes a Lokomotive cluster with Azure Arc.
+
+## Prerequisites
+
+1. Register providers for Azure Arc enabled Kubernetes:
+   ```bash
+   az provider register --namespace Microsoft.Kubernetes
+   az provider register --namespace Microsoft.KubernetesConfiguration
+   az provider register --namespace Microsoft.ExtendedLocation
+   ```
+
+   Monitor the registration process. Registration may take up to 10 minutes.
+
+   ```bash
+   az provider show -n Microsoft.Kubernetes -o table
+   az provider show -n Microsoft.KubernetesConfiguration -o table
+   az provider show -n Microsoft.ExtendedLocation -o table
+   ```
+
+   **NOTE**: This registration is needed only once per tenant.
+
+2. Create a resource group
+
+   Example:
+
+   ```bash
+   az group create --name AzureArcTest --location EastUS --output table
+   ```
+
+3. Register an application with the Microsoft identity platform
```

Comment:
Can we replace this whole section using `az` CLI instead of Azure Dashboard? That way it is copy-pastable.

Reply:
E.g. using `az ad sp create-for-rbac --role contributor` command like in https://github.com/kinvolk/clusterapi-flatcar/commit/922ea8a7e9eaf803affbcc7a60ff9d4e2002e9a7#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R47.